client_side/.infrastructure/verify_task.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+"""
+Determine whether the user's command has solved the task.
+
+The first input is the current true task code. The remaining inputs are the
+user's command.
+
+This script has the following exit status
+- Prints `success` to `stdout` if the actual output matches expected. Exit code
+  is `0`.
+- Prints `incomplete` to `stdout` if the actual output does not match expected.
+  - If the task is a file system task, exit code is `1`.
+  - If the task is a select task:
+    - If the file system has been changed, exit code is `2`.
+    - Otherwise, exit code is `3`.
+In addition, two files called "task_actual" and "task_expected" will be created
+in /tmp/ if the verification fails.
+"""
+
+import sys
+import os
+import shutil
+import subprocess
+import filecmp
+import tarfile
+
+# Gets all the environment variables and sets up the user output directory
+FS_DIR = os.environ['FS_DIR']
+
+USER_OUT_DIR = os.environ['USER_OUT']
+if not os.path.exists(USER_OUT_DIR):
+    os.mkdir(USER_OUT_DIR)
+
+# Establishes files for all the outputs
+USER_STDERR = os.path.join(USER_OUT_DIR, 'std_err')
+
+USER_FS_FILE = os.path.join(USER_OUT_DIR, 'fs_out')
+USER_STDOUT_FILE = os.path.join(USER_OUT_DIR, 'std_out')
+
+ACT_FILE = os.path.join('/tmp', 'task_actual')
+EXP_FILE = os.path.join('/tmp', 'task_expected')
+
+# There are two types of tasks: those that expect output, and
+# those that expect a modification to the file system.
+FILESYSTEM_TASKS = {'b', 'c', 'd', 'e', 'f', 'k', 'l', 'o', 'p', 't', 'v'}
+
+def main():
+    class cd:
+        """Context manager for changing the current working directory"""
+        def __init__(self, newPath):
+            self.newPath = os.path.expanduser(newPath)
+
+        def __enter__(self):
+            self.savedPath = os.getcwd()
+            os.chdir(self.newPath)
+
+        def __exit__(self, etype, value, traceback):
+            os.chdir(self.savedPath)
+
+    # the current task number, as a str
```

Comment:
This looks like the true task code

Reply:
Yes, this comment slipped through the cracks.

Reply:
Resolved in 8af6cc06a5d30b1f6a4ab6d5ef43f66d6cbe83df.